react-web/src/components/StoryPointer.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { PointSelector } from './PointSelector';
+import { PointerTable } from './PointerTable';
+import { Button } from 'patternfly-react';
+
+const POINTERS_EVENT = 'pointers';
+const SCORE_EVENT = 'score';
+const SHOW_EVENT = 'show';
+const CLEAR_EVENT = 'clear';
+
+const isNewPointerJoined = data => {
+  return data.event === POINTERS_EVENT;
+};
+
+const isScoreUpdated = data => {
+  return data.event === SCORE_EVENT;
+};
+
+const isShowPoints = data => {
+  return data.event === SHOW_EVENT;
+};
+
+const isClearPoints = data => {
+  return data.event === CLEAR_EVENT;
+};
+
+const createEvent = type => {
+  return {'event': type}
+};
+
+export class StoryPointer extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      pointers: []
+    };
+
+    this.pointSelected = this.pointSelected.bind(this);
+    this.showPoints = this.showPoints.bind(this);
+    this.clearPoints = this.clearPoints.bind(this);
+  }
+
+  componentDidMount() {
+    this.connectToSession(); 
+  }
+
+  componentWillUnmount() {
+    this.disconnectFromSession();
+  }
+
+  connectToSession() {
+    const url = `${process.env.REACT_APP_WS_HOST}/session/${this.props.sessionId}?name=${this.props.userName}`;
+    this.socket = new WebSocket(url);
+    this.socket.onopen = () => {
+      this.socket.onmessage = this.onMessage.bind(this);
+      this.socket.onerror = this.onError.bind(this);
+    };
+  }
+
+  disconnectFromSession() {
+    this.socket.close();
+  }
+
+  createUpdatedStoryPoints(data) {
+    let pointerIndex = this.state.pointers.findIndex(pointer => pointer.name === data.name);
+    const result = [...this.state.pointers];
+    if (pointerIndex > -1) {
+      let pointer = result[pointerIndex];
+      result[pointerIndex] = {
+        name: data.name,
+        score: data.score,
+        show: pointer.show
+      };
+    }
+    return result;
+  }
+
+  createPointersAsShowing() {
+    return this.state.pointers.map(pointer => Object.assign({}, pointer, {show: true}));
+  }
+
+  createClearedPointers() {
+    return this.state.pointers.map(pointer => Object.assign({}, pointer, {score: ''}));
```

Comment:
When the points are shown, there is no way of getting them set to hidden again. Can we include setting `show` to false here as well to reset it?

Reply:
:+1: